docs/DeveloperGuide.adoc

```diff
@@ -153,89 +164,250 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+- amount: `Price`, amount of budget limit
+- startDate: `Timestamp`, the start date of the budget (for the current period)
+- endDate: `Timestamp`, the end date of the budget (for the current period)
+- period: `Period`, the recurring period of the budget
+- expenses: `ObservableList<Expense>`, the list of all expenses under this budget
+- isPrimary: `boolean`, a flag indicating the primary status of this budget
+- proportionUsed: `Percentage`, the proportion of the total amount that has been used up
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+image::BudgetClassDiagram.png[]
 
-image::UndoRedoState5.png[]
+Each budget, once added, will recur infinitely. This is achieved by the refresh() method, which checks whether
+the budget has expired, and refreshes it to a new period if necessary, upon launching MooLah.
 
-The following activity diagram summarizes what happens when a user executes a new command:
+A budget is like a sliding window that moves in the positive direction of time axis. The window has a fixed size,
+which is the period of the budget. Only expenses within the current window will be shown in the app. Every time
+the budget refreshes, the window "slides" to the next period, starting with an empty screen that gradually gets
+populated with newly added expenses over time.
 
-image::CommitActivityDiagram.png[]
+A budget keeps a list of all expenses ever been tracked by this budget, including historical ones. Additionally,
+it has a method `getCurrentPeriodExpenses()` that generates a filtered list of expenses within the current budgeting
+period, which interacts with GUI to be shown in the current screen.
+
+The fundamental classes of the budget feature are:
+- `Budget` : main functional backend class
+- `JsonAdaptedBudget` : for storage of a `Budget` object in Json format
+- `UniqueBudgetList` : a list of non-duplicate budgets in MooLah
+- `AddBudgetCommand` and `SwitchBudgetCommand` : budget-related commands
+        `AddBudgetCommand` creates a new `Budget` and adds it to `UniqueBudgetList`
+        `SwitchBudgetCommand` switches the primary budget to the specified budget
+- `AddBudgetCommandParser` and `SwitchBudgetCommandParser` : for parsing CLI input into commands
+
+image::AddBudgetCommandClassDiagram.png[]
+
+image::SwitchBudgetCommandClassDiagram.png[]
+
+Additionally, util classes like `Percentage` which wraps a percentage for easier calculation and validation of proportionUsed.
+
+image::PercentageClassDiagram.png[]
+
+The following sequence diagram shows how the add budget operation works:
+
+image::AddBudgetSequenceDiagram.png[]
```

Comment:
Is there still a class called "AddressBookParser" and "AddressBook"? I believe with the refactor they should not be present anymore

the last method call in the diagram:
> setPrimary(toAdd)

is an object calling a method of itself. It is of the wrong notation.